Packs/Palo_Alto_Networks_Enterprise_DLP/Integrations/Palo_Alto_Networks_Enterprise_DLP/Palo_Alto_Networks_Enterprise_DLP.py

```diff
@@ -405,49 +473,59 @@ def is_reset_triggered():
     return False
 
 
-def get_base_url():
-    return BASE_URL
+def fetch_notifications(client: Client, regions: str):
+    integration_context = demisto.getIntegrationContext()
+    access_token = integration_context.get(ACCESS_TOKEN)
+    if access_token:
+        client.set_access_token(access_token)
+
+    incidents = fetch_incidents(
+        client=client,
+        regions=regions
+    )
+    print_debug_msg(f"Received {len(incidents)} incidents")
+    if not is_reset_triggered():
+        demisto.createIncidents(incidents)
+        new_ctx = {
+            ACCESS_TOKEN: client.access_token,
+            'samples': incidents
+        }
+        demisto.setIntegrationContext(new_ctx)
+    elif len(incidents) > 0:
+        print_debug_msg(f"Skipped {len(incidents)} incidents because of reset")
 
 
-def long_running_execution_command(params: Dict):
+def long_running_execution_command(client: Client, params: Dict):
     """
     Long running execution of fetching incidents from Palo Alto Networks Enterprise DLP.
     Will continue to fetch in an infinite loop.
     Args:
         params (Dict): Demisto params.
 
     """
+    demisto.setIntegrationContext({ACCESS_TOKEN: ''})
     regions = demisto.get(params, 'dlp_regions', '')
-    refresh_token = params.get('refresh_token')
-    url = get_base_url()
+    sleep_time = FETCH_SLEEP
+    last_time_sleep_interval_queries = math.floor(datetime.now().timestamp())
     while True:
         try:
-            integration_context = demisto.getIntegrationContext()
-            access_token = integration_context.get(ACCESS_TOKEN)
-            access_token = params.get('access_token') if not access_token else access_token
-            client = Client(url, refresh_token, access_token, params.get('insecure'), params.get('proxy'))
-            incidents = fetch_incidents(
-                client=client,
-                regions=regions
-            )
-            print_debug_msg(f"Received {len(incidents)} incidents")
-            if not is_reset_triggered():
-                demisto.createIncidents(incidents)
-                new_ctx = {
-                    ACCESS_TOKEN: client.access_token,
-                    'samples': incidents
-                }
-                demisto.setIntegrationContext(new_ctx)
-            elif len(incidents) > 0:
-                print_debug_msg(f"Skipped {len(incidents)} incidents because of reset")
+            current_time = math.floor(datetime.now().timestamp())
+            fetch_notifications(client, regions)
+
+            if current_time - last_time_sleep_interval_queries > 5 * 60:
+                overriden_sleep_time = client.query_for_sleep_time()
+                last_time_sleep_interval_queries = current_time
+                if overriden_sleep_time:
+                    print_debug_msg(f'Setting sleep time to value from backend: {overriden_sleep_time}')
+                    sleep_time = overriden_sleep_time
```

Comment:
Since we are pulling incidents every 5 seconds in this integration, our backend team is concerned about the performance impact it will have on our servers. They created an API for me to call every 5 minutes. This API will tell me how much time the integration should rest between each pull. If the server is healthy, it doesn't return anything. Otherwise it will return the # of seconds to rest. That is the reason for the if statement.